packages/jsx/src/element/is-keyed-element.ts

```diff
@@ -0,0 +1,9 @@
+import type { TSESTree } from "@typescript-eslint/types";
+import { AST_NODE_TYPES } from "@typescript-eslint/types";
+
+import { isJSXAttributeKey } from "../attribute";
+
+export function isKeyedElement(node: TSESTree.Node) {
+  return node.type === AST_NODE_TYPES.JSXElement
+    && node.openingElement.attributes.some(isJSXAttributeKey);
```

Comment:
This implementation takes too few cases into account, this case should use `hasProp` which calls `findPropInAttributes` in the same package.
Also see: https://github.com/jsx-eslint/eslint-plugin-react/issues/1753

Later, I will add a `no-spreading-key` to `@eslint-react/eslint-plugin-jsx` since it is proposed to be deprecated in the https://github.com/reactjs/rfcs/pull/107 RFC.